src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -847,12 +845,59 @@ public void shouldNotExtendLoanDueDateIfOverdueLoanIsRecalled() {
     ));
   }
 
+  @Test
+  public void shouldExtendLoanDueDateIfOverdueLoanIsRecalledAndConditionsAreMet() {
```

Comment:
What conditions need to be met?

Reply:
Which scenario from the story does this demonstrate?

Reply:
This addresses Scenario 1. It seemed to us that the other scenarios have been covered by existing tests.

Reply:
As for the name, it seemed that a concise expression of the specific conditions was elusive. If you have a suggestion I will gladly change it.

Reply:
What do you think to the following? 

If extension must be allowed for the alternate period to be defined then I think the `AndPolicyAllowsExtension` part can be dropped.

```suggestion
  public void shouldExtendLoanDueDateByAlternatePeriodWhenOverdueLoanIsRecalledAndPolicyAllowsExtension() {
```

Reply:
> It seemed to us that the other scenarios have been covered by existing tests.

Which test covers scenario 2 given that is new behaviour according to the story?

Reply:
Given that scenario 3 now makes the configuration to disallow extension explicit, I think the test `shouldNotExtendLoanDueDateIfOverdueLoanIsRecalled` should be changed to reflect that.